lib/pos_mempool_test.go

```diff
@@ -0,0 +1,323 @@
+		},
+	)
+}
+
+func TestTimeBucketBasic(t *testing.T) {
+	seed := int64(33)
+	testCases := 1000
+	indexRange := 100
+	timestampRange := uint64(10000)
+
+	require := require.New(t)
+	rand := rand.New(rand.NewSource(seed))
+	globalParams := _testGetDefaultGlobalParams()
+	randomIndex := rand.Intn(indexRange)
+	baseRate, bucketMultiplier := globalParams.computeFeeBucketBaseAndMultiplier()
+	feeMin, feeMax := ComputeTimeBucketRangeFromIndex(randomIndex, baseRate, bucketMultiplier)
+	txnPool := _testGetRandomMempoolTxns(rand, feeMin, feeMax, timestampRange, testCases)
+
+	// Create new FeeBucket and add the txn pool
+	timeBucket := NewTimeBucket(randomIndex, baseRate, bucketMultiplier)
+	_testBucketStandardAddTest(t, txnPool, globalParams, true,
+		func(tx *MempoolTx) {
+			require.Equal(nil, timeBucket.AddTransaction(tx))
+		},
+		func() []*MempoolTx {
+			return timeBucket.GetTransactions()
+		},
+	)
+}
+
+func _testGetDefaultGlobalParams() *GlobalParamsEntry {
+	globalParams := InitialGlobalParamsEntry
+	globalParams.FeeBucketBaseRateNanosPerKB = 1000
+	globalParams.FeeBucketRateMultiplierBasisPoints = 1000
+
+	return &globalParams
+}
+
+func _testGetRandomMempoolTxns(rand *rand.Rand, feeMin uint64, feeMax uint64, timestampRange uint64, numTxns int) []*MempoolTx {
+	txnPool := []*MempoolTx{}
+	for ii := 0; ii < numTxns; ii++ {
+		txnPool = append(txnPool, &MempoolTx{
+			FeePerKB: rand.Uint64()%(feeMax-feeMin) + feeMin,
+			Added:    time.UnixMicro(int64(rand.Uint64() % timestampRange)),
+			Hash:     NewBlockHash(RandomBytes(32)),
+		})
+	}
+	return txnPool
+}
+
+func _testMapMempoolTxnToTimeBucket(txn *MempoolTx, globalParams *GlobalParamsEntry) *FeeTimeBucket {
+	baseRate, bucketMultiplier := globalParams.computeFeeBucketBaseAndMultiplier()
+	timeBucketIndex := ComputeTimeBucketIndexFromFeePerKBNanos(txn.FeePerKB, baseRate, bucketMultiplier)
+	return NewTimeBucket(timeBucketIndex, baseRate, bucketMultiplier)
+}
+
+func _testSortMempoolTxnsByFeeTime(txnPool []*MempoolTx, globalParams *GlobalParamsEntry, timeOnly bool) []*MempoolTx {
+	if !timeOnly {
+		sort.Slice(txnPool, func(i, j int) bool {
+			timeBucketI := _testMapMempoolTxnToTimeBucket(txnPool[i], globalParams)
+			timeBucketJ := _testMapMempoolTxnToTimeBucket(txnPool[j], globalParams)
+
+			feeComparison := FeeTimeBucketComparator(timeBucketI, timeBucketJ)
+			if feeComparison == 1 {
+				return false
+			} else if feeComparison == -1 {
+				return true
+			} else {
+				timeComparison := MempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
+				if timeComparison == 1 {
+					return false
+				} else if timeComparison == -1 {
+					return true
+				} else {
+					return true
+				}
+			}
+
+		})
+	} else {
+		sort.Slice(txnPool, func(i, j int) bool {
+			timeComparison := MempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
+			if timeComparison == 1 {
+				return false
+			} else if timeComparison == -1 {
+				return true
+			} else {
+				return true
+			}
+		})
+	}
+	return txnPool
```

Comment:
we can simplify this one a lot.
```suggestion
	return txnPool
	}
	sort.Slice(txnPool, func(i, j int) bool {
		timeComparison := MempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
		return timeComparison != 1
	})
	return txnPool
```